### DISCUSSION TITLE
Remove self reference when fetching self relations

### DISCUSSION THREAD

**Discussion by Darren120** [prisma/prisma#10033] [Q&A]
Query : 

```
await this.prisma.user.findMany({
      where : {Id : 1},
     include : {SubscribedBy:true, SubscribedTo: true}
      
    
    } )
```
Result:
```
[
  {
    "Id": 1,
    "CreatedAt": "2021-10-30T23:54:36.061Z",
    "DeletedAt": null,
    "UpdatedAt": "2021-10-30T23:54:36.062Z",
    "SubscribedBy": [
      {
        "SubscriberId": 2,
        "SubscribedId": 1
      },
      {
        "SubscriberId": 3,
        "SubscribedId": 1
      }
    ],
    "SubscribedTo": [
      {
        "SubscriberId": 1,
        "SubscribedId": 2
      }
    ]
  }
]
```

Remove self relation references from being in final result unless specificed. So in this case, in SubscribedBy, the SubscribedId should be omitted, and in SubscribedTo, the SubscriberId should be omitted.

Its so confusing, the more i read it the more confused I get.

it gets even more confusing when I use terms like Following and Follower. I can omit them using select but I hope you guys can add a way to omit self reference id by default

**Comment by austincrim** [comment]
Hi @Darren120 👋,

Like you already mentioned, the way to omit these IDs would be to use a nested select. Could you open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) specifying your exact use case and context?

Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
